### ISSUE TITLE
Access log token discovery or validation for xDS

### ISSUE THREAD

**Issue by jpeach** [envoyproxy/envoy#13620]
*Description*:

An xDS server may need to deal with multiple Envoy versions (either simultaneously, or over its deployed lifetime). The set of valid access log tokens can vary across Envoy versions. If the xDS server sends a configuration with an invalid log token, that update is rejected with a `Not supported field in StreamInfo` error message. Other than by parsing the message string in the `ErrorDetail`, the xDS server doesn't have any way to know that the log token was invalid.

One way to solve this would be for xDS servers to have access to a canonical list of valid log tokens for the Envoy version. Another way could be to use the `ErrorDetail.Details` field to expose some structured information that the xDS server could use to drop the invalid log token.

**Comment by htuch** [comment]
Yeah, I think graceful handling is the way to go. We could model this as client capabilities, but the API principle is to reserve them for when they are really needed, when there is no graceful degradation option.
